crates/ruff_linter/src/rules/flake8_bugbear/rules/assert_raises_exception.rs

```diff
@@ -73,33 +106,31 @@ pub(crate) fn assert_raises_exception(checker: &Checker, items: &[WithItem]) {
             continue;
         }
 
-        let [arg] = &*arguments.args else {
-            continue;
-        };
-
-        let semantic = checker.semantic();
-
-        let Some(builtin_symbol) = semantic.resolve_builtin_symbol(arg) else {
-            continue;
-        };
-
-        let exception = match builtin_symbol {
-            "Exception" => ExceptionKind::Exception,
-            "BaseException" => ExceptionKind::BaseException,
-            _ => continue,
-        };
-
-        if !(matches!(func.as_ref(), Expr::Attribute(ast::ExprAttribute { attr, .. }) if attr == "assertRaises")
-            || semantic
-                .resolve_qualified_name(func)
-                .is_some_and(|qualified_name| {
-                    matches!(qualified_name.segments(), ["pytest", "raises"])
-                })
-                && arguments.find_keyword("match").is_none())
+        if let Some(exception) =
+            detect_blind_exception(checker.semantic(), func.as_ref(), arguments)
         {
-            continue;
+            checker.report_diagnostic(AssertRaisesException { exception }, item.range());
         }
+    }
+}
+
+/// B017 (call form)
+pub(crate) fn assert_raises_exception_call(
+    checker: &Checker,
+    ast::ExprCall {
+        func,
+        arguments,
+        range: _,
+        node_index: _,
+    }: &ast::ExprCall,
+) {
+    let semantic = checker.semantic();
+
+    if arguments.args.len() < 2 && arguments.find_argument("func", 1).is_none() {
+        return;
+    }
 
-        checker.report_diagnostic(AssertRaisesException { exception }, item.range());
+    if let Some(exception) = detect_blind_exception(semantic, func.as_ref(), arguments) {
+        checker.report_diagnostic(AssertRaisesException { exception }, func.range());
```

Comment:
I think this is supposed to be the `call` range to match the `with` version.

```suggestion
        checker.report_diagnostic(AssertRaisesException { exception }, range);
```

with `range` from the `ExprCall` above.